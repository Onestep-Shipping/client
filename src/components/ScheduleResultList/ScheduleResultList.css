.container-types-quote {
  width: 90%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  line-height: 23px;
}

.quote-body {
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: space-between;
  margin-bottom: 20px;
}