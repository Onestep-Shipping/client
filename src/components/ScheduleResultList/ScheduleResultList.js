import React, { useState, useContext } from 'react';
import './ScheduleResultList.css';
import PropTypes from 'prop-types';
import FixedSizeList from '../FixedSizeList/FixedSizeList.js';
import { AuthContext } from '../../context/AuthContext.js';
import arrowIcon from '../../assets/arrow-down.svg';
import Utils from '../../utils/Helpers.js';

const ScheduleResultList = props => {
  const { action, scheduleList, remove = { _id: null } } = props;
  const [validity, setValidity] = useState(0);
  const { currentUser, isAdmin } = useContext(AuthContext);

  const RESULT_HEADERS = ['#', 'Port of Loading', 'Transshipments',
    'Vessels / Services', 'Port of Discharge', 'Transit Time'];
  
  const QUOTE_HEADERS = ['Ocean Freight (All-in)',
    'Documentation Fee', 'Administration Fee'];

  const [currentBookingIndex, setCurrentBookingIndex] = useState(0);

  const row = (schedule, ind) => {
    const quotes =
      schedule.route.quoteHistory
        .filter(quote => Date.parse(quote.validity.startDate) <=  Date.parse(schedule.startDate))
        .slice(0, 2)
    return (
      <div className="schedule-result-row-container" key={ind}>
        <div
          className={'schedule-result-row' +
            (ind === currentBookingIndex ? '-selected' : '')}
          onClick={() => setCurrentBookingIndex(ind)}>
          <div className="col-numb">
            <text className="schedule-result-text">{ind + 1}</text>
          </div>
          <div className="col">
            <text className="schedule-result-text">{schedule.route.startLocation}</text>
            <text className="schedule-result-text-time">
              {Utils.formatISOString(schedule.startDate)}
            </text>
          </div>
          <div className="col">
            <text className="schedule-result-text">{schedule.transshipment}</text>
          </div>
          <div className="col">
            <text className="schedule-result-text">{schedule.vessels}</text>
          </div>
          <div className="col">
            <text className="schedule-result-text">{schedule.route.endLocation}</text>
            <text className="schedule-result-text-time">
              {Utils.formatISOString(schedule.endDate)}
            </text>
          </div>
          <div className="col">
            <text className="schedule-result-text">{schedule.transitTime}</text>
          </div>
        </div>

        {(ind === currentBookingIndex && !isAdmin && currentUser) &&
        <div className="quote-dropdown">
          <div className="validity-container">
            <text className="schedule-result-header-text">
              {validity === 0 ? "CURRENT" : "UPCOMING"} VALIDITY: 
            </text>
            {Utils.formatValidity(quotes[validity].validity)}
            {quotes > 1 && 
            <img 
              className={"arrow-icon-point-" + (validity === 0 ? "right" : "left")} 
              src={arrowIcon} alt="Arrow Icon" 
              onClick={() => setValidity(validity === 0 ? 1 : 0)} 
            />}
          </div>
          <div className="quote-body">
            <div className="col3">
              <text className="schedule-result-header-text">{QUOTE_HEADERS[0].toUpperCase()}</text>
              {quotes[validity].selling.oceanFreight.map((container, ind) => 
                <div key={ind} className="container-types-quote">
                  <text className="schedule-result-text">{container.containerType}: </text>
                  <text className="schedule-result-text">${Utils.comma(container.price)}</text>
                </div>
              )}
            </div>
            <div className="col3">
              <text className="schedule-result-header-text">{QUOTE_HEADERS[1].toUpperCase()}</text>
              <text className="schedule-result-text">${Utils.comma(quotes[validity].selling.docFee)}</text>
            </div>
            <div className="col3">
              <text className="schedule-result-header-text">{QUOTE_HEADERS[2].toUpperCase()}</text>
              <text className="schedule-result-text">${Utils.comma(quotes[validity].selling.adminFee)}</text>
            </div>
          </div>
          {quotes[validity].except &&
          <div className="finance-display-form">
            Note: FAK Rates (Except {quotes[validity].except})
          </div>}
          <button className="result-button" onClick={() => action(schedule, quotes[0])}>
            Accept
          </button>
        </div>}
      </div>
    );
  }

  return (
    <div className="schedule-result-container">
      <FixedSizeList 
        headers={RESULT_HEADERS} 
        data={scheduleList.filter(item => item._id !== remove._id)} 
        row={row}
      />
    </div>
  );
};

export default ScheduleResultList;

ScheduleResultList.propTypes = {
  scheduleList: PropTypes.array,
  action: PropTypes.func,
  remove: PropTypes.object,
};