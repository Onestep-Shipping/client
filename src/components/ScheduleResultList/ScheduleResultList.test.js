import AuthProvider from '../../context/AuthContext.js';
import React from 'react';
import ScheduleResultList from './ScheduleResultList';
import { shallow } from 'enzyme';

const mockSubmit = jest.fn();

let wrapper = shallow( 
  <AuthProvider>
    <ScheduleResultList 
      action={mockSubmit} 
      scheduleList={[]} />
  </AuthProvider>
);

describe('ScheduleResultList', () => {
  it('should render the ScheduleResultList Component correctly', () => {   
    expect(ScheduleResultList).toMatchSnapshot();
  });
});