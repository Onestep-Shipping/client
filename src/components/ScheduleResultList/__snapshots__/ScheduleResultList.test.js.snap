// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleResultList should render the ScheduleResultList Component correctly 1`] = `[Function]`;
